react/modules/useCumulativeHeightState.ts

```diff
@@ -0,0 +1,37 @@
+import { useState } from 'react'
+
+type State = Record<number, number>
+
+const useCumulativeHeightState = () => {
+  const [state, set] = useState<State>({})
+
+  const updateRowHeight = ({
+    height,
+    index,
+  }: {
+    height: number
+    index: number
+  }) => {
+    if (state[index] === height) {
+      return
+    }
+
+    set(prev => ({ ...prev, [index]: height }))
+  }
+
+  const getAccumulatedHeight = (index: number) => {
+    const sortedIndices = Object.keys(state)
```

Comment:
I am not sure if "Indices" is right. In case it is not, use "Indexes" instead.

Reply:
Done 🙏